l10n_it_account_balance_eu/wizards/account_balance_eu_wizard.py

```diff
@@ -0,0 +1,157 @@
+    amount = fields.Float(string="Amount", readonly=True)
+
+
+class CreateBalanceWizard(models.TransientModel):
+    _name = "account.balance.eu.wizard"
+    _description = "Wizard for balance EU calculation"
+
+    def _default_date_range(self):
+        return (
+            self.env["date.range"]
+            .search([("name", "=", str(date.today().year - 1))])
+            .id
+        )
+
+    def _default_date_from(self):
+        return date(date.today().year - 1, 1, 1)
+
+    def _default_date_to(self):
+        return date(date.today().year - 1, 12, 31)
+
+    date_range_id = fields.Many2one(
+        comodel_name="date.range",
+        string="Date range",
+        default=_default_date_range,
+    )
+    date_from = fields.Date(
+        string="Balance from date", required=True, default=_default_date_from
+    )
+    date_to = fields.Date(
+        string="Balance to date", required=True, default=_default_date_to
+    )
+    values_precision = fields.Selection(
+        [
+            ("d", "2 decimals Euro"),
+            ("u", "euro units"),
+        ],
+        string="Values show as",
+        default="d",
+        required=True,
+    )
+    hide_acc_amount_0 = fields.Boolean(
+        string="Hide account with amount 0", default=True
+    )
+    only_posted_move = fields.Boolean(
+        string="Use only posted registration", default=True
+    )
+    ignore_closing_move = fields.Boolean(
+        string="Ignore closing registration", default=True
+    )
+    log_warnings = fields.Text(string="WARNING:", default="")
+    # BALANCE HEADER
+    company_id = fields.Many2one("res.company", string="Company")
+    currency_id = fields.Many2one("res.currency", string="Currency")
+    name = fields.Char(string="Name", compute="_compute_period_data")
+    year = fields.Integer(string="Year", compute="_compute_period_data")
+    # COMPANY DATA
+    company_name = fields.Char(string="Company Name")
+    address = fields.Char(string="Address")
+    city = fields.Char(string="City")
+    rea_office = fields.Char(string="REA office")
+    rea_num = fields.Char(string="REA number")
+    rea_capital = fields.Float(string="Share Capital")
+    fiscalcode = fields.Char(string="Fiscal Code")
+    vat_code = fields.Char(string="VAT number")
+    vat_code_nation = fields.Char(string="VAT number nation")
+    balance_log_ids = fields.One2many(
+        "account.balance.eu.log",
+        "balance_id",
+        string="Unlinked Account",
+        auto_join=True,
+    )
+    state = fields.Selection(
+        [
+            ("OK", "COMPLETE"),
+            ("UNLINKED_ACCOUNTS", "CHECK ACCOUNTS"),
+            ("UNBALANCED", "UNBALANCED"),
+        ],
+        string="State",
+        default="OK",
+        readonly=True,
+    )
+
+    @api.onchange("date_range_id")
+    def onchange_date_range_id(self):
+        """Handle date range change."""
+        self.date_from = self.date_range_id.date_start
+        self.date_to = self.date_range_id.date_end
```

Comment:
Quando `date_range_id` è vuoto, questo `onchange` svuota i campi `date_from` e `date_to` che avrebbero invece i loro default aggiunti dalle ultime modifiche.
Non so se è voluto, ma ti consiglio di modificare le date solo quando è stato selezionato un Date Range:
```suggestion
        date_range = self.date_range_id
        if date_range:
            self.date_from = date_range.date_start
            self.date_to = date_range.date_end
```

Reply:
fatto